widget/embedded/src/components/SwapDetails/SwapDetails.helpers.tsx

```diff
@@ -0,0 +1,215 @@
+import type { getSwapMessages } from '../../utils/swap';
+import type {
+  PendingSwap,
+  PendingSwapStep,
+} from '@rango-dev/queue-manager-rango-preset';
+import type {
+  PriceImpactWarningLevel,
+  Step,
+  StepDetailsProps,
+} from '@rango-dev/ui';
+import type { ConnectResult } from '@rango-dev/wallets-react';
+import type BigNumber from 'bignumber.js';
+
+import { i18n } from '@lingui/core';
+import { PendingSwapNetworkStatus } from '@rango-dev/queue-manager-rango-preset';
+import { Alert, Button, InfoIcon, LinkIcon, Tooltip } from '@rango-dev/ui';
+import React, { Fragment } from 'react';
+
+import { numberToString } from '../../utils/numbers';
+import { isNetworkStatusInWarningState } from '../../utils/swap';
+
+import { Alerts } from './SwapDetails.styles';
+
+export const LOW_PRICE_IMPACT = 3;
+export const HIGHT_PRICE_IMPACT = 10;
+export const RESET_INTERVAL = 2_000;
+export const MIN_DECIMALS = 0;
+export const AMOUNT_DECIMALS = 4;
+export const PERCENT_DECIMALS = 2;
+export const SECONDS = 60;
+
+export const getSteps = (
+  pendingSwap: PendingSwap,
+  switchNetwork: (() => Promise<ConnectResult>) | undefined,
+  showNetworkModal: PendingSwapNetworkStatus | null | undefined,
+  setNetworkModal: (network: PendingSwapNetworkStatus) => void,
+  message: ReturnType<typeof getSwapMessages>
+): Step[] => {
+  return pendingSwap.steps.map((step, index) => {
+    return {
+      from: {
+        token: { displayName: step.fromSymbol, image: step.fromLogo },
+        chain: {
+          displayName: step.fromBlockchain,
+          image: step.fromBlockchainLogo,
+        },
+        price: {
+          value:
+            index === 0
```

Comment:
Please rewrite it like (`numberToString(amountToConvert, AMOUNT_DECIMALS, AMOUNT_DECIMALS)`) to avoid duplicated codes:

```
const amountToConvert =
  index === 0
    ? pendingSwap.inputAmount
    : pendingSwap.steps[index - 1].outputAmount;
```

Reply:
fixed.

Reply:
fixed.